smartsim/_core/launcher/step/dragonStep.py

```diff
@@ -90,3 +100,157 @@ def _get_exe_args_list(run_setting: DragonRunSettings) -> t.List[str]:
+        :type batch_settings: SbatchSettings
+        """
+        super().__init__(name, cwd, batch_settings)
+        self.steps: t.List[Step] = []
+        self.managed = True
+        self.batch_settings = batch_settings
+        self._request_file_name = "requests.json"
+
+    def get_launch_cmd(self) -> t.List[str]:
+        """Get the launch command for the batch
+
+        :return: launch command for the batch
+        :rtype: list[str]
+        """
+        if isinstance(self.batch_settings, SbatchSettings):
+            script = self._write_sbatch_script()
+            return [self.batch_settings.batch_cmd, "--parsable", script]
+        if isinstance(self.batch_settings, QsubBatchSettings):
+            script = self._write_qsub_script()
+            return [self.batch_settings.batch_cmd, script]
+
+        raise SSUnsupportedError(
+            "DragonBatchStep only support SbatchSettings and QsubBatchSettings"
+        )
+
+    def add_to_batch(self, step: Step) -> None:
+        """Add a job step to this batch
+
+        :param step: a job step instance e.g. DragonStep
+        :type step: Step
+        """
+        self.steps.append(step)
+        logger.debug(f"Added step command to batch for {step.name}")
+
+    @staticmethod
+    def _dragon_entrypoint_cmd(request_file: str) -> str:
+        """Return command needed to run the Dragon entrypoint"""
+        cmd = [
+            f"{sys.executable}",
+            "-m",
+            "smartsim._core.entrypoints.dragon_client",
+            "+submit",
+            f"{request_file}",
+            "\n",
+        ]
+        return " ".join(cmd)
+
+    def _write_request_file(self) -> str:
+        """Write json file with requests to submit to Dragon server"""
+        request_file = self.get_step_file(
+            ending="json", script_name=self._request_file_name
+        )
+        requests = []
+        for step in self.steps:
+            run_settings = t.cast(DragonRunSettings, step.step_settings)
+            run_args = run_settings.run_args
+            env = run_settings.env_vars
+            nodes = int(run_args.get("nodes", None) or 1)
+            tasks_per_node = int(run_args.get("tasks-per-node", None) or 1)
+
+            cmd = step.get_launch_cmd()
+            out, err = step.get_output_files()
+            request = DragonRunRequest(
+                exe=cmd[0],
+                exe_args=cmd[1:],
+                path=step.cwd,
+                name=step.name,
+                nodes=nodes,
+                tasks_per_node=tasks_per_node,
+                env=env,
+                current_env=os.environ,
+                output_file=out,
+                error_file=err,
+            )
+            requests.append(request_registry.to_string(request))
+        with open(request_file, "w", encoding="utf-8") as script_file:
+            script_file.write(json.dumps(requests))
+
+        return request_file
+
+    def _write_sbatch_script(self) -> str:
```

Comment:
consider replacing these `_write_xxx_script` functions w/template loading. It could beccome something more akin to the following pseudocode:

```
def write_batch_script(launcher: str, request_file: ???) -> str:
  # load correct template, e.g. /foo/bar/{launcher}/batch.tpl
  template = get_batch_template(launcher)
  script = template.render(request_file)
  return script
```

Reply:
I think that's something to do for all `Step` classes at the same time, as I would like to keep divergence to the minimum.